Kuzzle/Kuzzle.cs

```diff
@@ -42,7 +43,7 @@ public interface IKuzzleApi {
     /// </summary>
     /// <returns>The query response.</returns>
     /// <param name="query">Kuzzle API query</param>
-    Task<Response> QueryAsync(JObject query);
+    Task<Response> QueryAsync(JObject query, QueryOptions options = null);
```

Comment:
It's worse than a global utility class: this new signatures tells that a QueryOptions object can be passed to all queries but the handled option values are only useful for several API routes only. 

First: the `notify` option doesn't need a utility option, just pass it in the JSON body of the actions handling that option.
Second: the only option that requires a dedicated function because the SDK has a different way to handle it than the API, is the `waitForRefresh` one. So all you need is your previous "HandleRefreshOption" function in a dedicated namespace and that is all. It's still a utility function, but instead of being made available in a static object global to the project, it's neatly stored in a dedicated and meaningful namespace.

Reply:
@scottinet @Aschen @benoitvidis It seems that your not all agree on what has to be done for this case, it's easy to change really quick but everyone seems to have a different opinion on that.

So I will let that has it is for now.

Reply:
I am sorry, I must have missed something, but why do we need a utility function at all?
Can't we just set the `waitForRefresh` like we do for `notify` from the controllers, and in `queryAsync`, tweak the json with something like this? :

```csharp
[..]
      if (query["waitForRefresh"] != null) {
        if (query["waitForRefresh"].ToObject<bool>()) {
          query["refresh"] = "wait_for";
        }
        query["waitForRefresh"].Remove();
      }
[..]
```

Reply:
I guess this is a possibility, but my first reaction is to disagree with it, because we're letting controllers set what we know is an unsupported option, to let the QueryAsync correct it afterward.

That said... I didn't think of that approach, and might have its usefulness depending on how we plan to handle the refresh option in the future.
I'm confident that we'll change it to a waitForRefresh boolean option, as this makes a lot more sense for users, and is more on par with the remaining of our API.

So... if we do that, and if we drop the support for the Kuzzle API v1 in this SDK, then we'd simply need to remove this snippet, instead of changing all methods to get rid of that utility function.

At that point, I guess it's up to @Shiranuit.

Reply:
So, I've changed everything another time to use the solution of @benoitvidis that I think is better and avoid massive modification to remove the utility method.